packages/next/src/index.ts

```diff
@@ -2720,7 +2720,18 @@ async function getServerlessPages(params: {
   for (const edgeFunctionFile of Object.keys(
     middlewareManifest?.functions ?? {}
   )) {
-    const edgePath = (edgeFunctionFile.slice(1) || 'index') + '.js';
+    let fileName = edgeFunctionFile.slice(1) || 'index';
+    // when using app directory the index page will have path /page
+    if (
+      fileName === 'page' &&
+      // `page` is index path only when using app directory
+      middlewareManifest?.functions?.[fileName + '.js']?.files?.some(
+        (file: string) => file.includes('/app/')
```

Comment:
I'm not sure this is the right way to detect App Router.

For example, what if a file was named `/components/app/file.js`?

I think you want to look in `normalizedAppPaths`.

cc @ijjk

Reply:
I am now using the function name to detect if it’s using app dir